client.go

```diff
@@ -529,6 +532,11 @@ func (client *client) RefreshCoordinator(consumerGroup string) error {
 // in the brokers map. It returns the broker that is registered, which may be the provided broker,
 // or a previously registered Broker instance. You must hold the write lock before calling this function.
 func (client *client) registerBroker(broker *Broker) {
+	if client.brokers == nil {
```

Comment:
this is a race condition, no?

It checks for client.brokers != nil
and then uses client.brokers

it could have been set to nil between the `if client.brokers == nil` and `if client.brokers[broker.ID()] == nil `